build/dockerfiles/dev.Dockerfile

```
@@ -6,68 +6,34 @@
-                libX11-devel libxkbcommon krb5-devel bash tar gzip rsync patch pkg-config glib2-devel coreutils-single glibc-minimal-langpack which util-linux-user --releasever 8 --nodocs -y && yum --installroot /mnt/rootfs clean all
-# node-gyp will search for python
-RUN cd /mnt/rootfs && ln -s /usr/bin/python3 ./usr/bin/python
-                
-RUN rm -rf /mnt/rootfs/var/cache/* /mnt/rootfs/var/log/dnf* /mnt/rootfs/var/log/yum.*
-
-# Download nodejs required by VS Code
-RUN mkdir -p /mnt/rootfs/opt/nodejs && curl -sL https://nodejs.org/download/release/v18.16.1/node-v18.16.1-linux-x64.tar.gz | tar xzf - -C /mnt/rootfs/opt/nodejs --strip-components=1
-
-# Download kubectl
-RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && \
-    chmod +x kubectl && \
-    mv kubectl /mnt/rootfs/usr/bin/
-
-# setup home folder inside the new fs
-ENV HOME=/mnt/rootfs/home/che
-ENV NPM_CONFIG_PREFIX=/mnt/rootfs/home/che/.npm-global
-ENV PATH /mnt/rootfs/opt/nodejs/bin:/mnt/rootfs/home/che/.npm-global/bin/:$PATH
-
-# install yarn
-RUN npm install -g yarn
+FROM quay.io/devfile/universal-developer-image:latest
 
-ENV ELECTRON_SKIP_BINARY_DOWNLOAD=1 \
-    PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=0
- 
-RUN cat /mnt/rootfs/etc/passwd | sed s#root:x.*#root:x:\${USER_ID}:\${GROUP_ID}::\${HOME}:/bin/bash#g > /mnt/rootfs/home/che/.passwd.template \
-    && cat /mnt/rootfs/etc/group | sed s#root:x:0:#root:x:0:0,\${USER_ID}:#g > /mnt/rootfs/home/che/.group.template
-
-COPY --from=machine-exec --chown=0:0 /go/bin/che-machine-exec /mnt/rootfs/bin/machine-exec
-COPY --chmod=755 /build/scripts/entrypoint-dev.sh /mnt/rootfs/entrypoint.sh
-COPY --chmod=755 /build/conf/dev/.p10k.zsh /mnt/rootfs/home/che/.p10k.zsh
-
-RUN mkdir -p /mnt/rootfs/projects
-
-RUN for f in "/mnt/rootfs/home/che" "/mnt/rootfs/etc/passwd" "/mnt/rootfs/etc/group" "/mnt/rootfs/projects" ; do\
-           chgrp -R 0 ${f} && \
-           chmod -R g+rwX ${f}; \
-       done
-
-FROM scratch
-COPY --from=ubi-micro-build /mnt/rootfs/ /
-ENV HOME=/home/che
-ENV NPM_CONFIG_PREFIX=/home/che/.npm-global
-ENV PATH /opt/nodejs/bin:/home/che/.npm-global/bin/:$PATH
-ENTRYPOINT /entrypoint.sh
-WORKDIR /projects
-RUN wget https://github.com/robbyrussell/oh-my-zsh/raw/master/tools/install.sh -O - | zsh && \
+USER 0
+
+RUN dnf -y install libsecret libX11-devel libxkbcommon \
+    "https://rpmfind.net/linux/centos/8-stream/BaseOS/x86_64/os/Packages/libsecret-devel-0.18.6-1.el8.x86_64.rpm" \
+    "https://rpmfind.net/linux/centos/8-stream/AppStream/x86_64/os/Packages/libxkbfile-1.1.0-1.el8.x86_64.rpm" \
+    "https://rpmfind.net/linux/centos/8-stream/PowerTools/x86_64/os/Packages/libxkbfile-devel-1.1.0-1.el8.x86_64.rpm" \
+    "https://rpmfind.net/linux/centos/8-stream/BaseOS/x86_64/os/Packages/zsh-5.5.1-6.el8_1.2.x86_64.rpm" \
+    util-linux-user
+
+# cleanup dnf cache
+RUN dnf -y clean all --enablerepo='*'
```

To reduce the image size it's better to install and clean in the same command otherwise we are going to commit a layer with all the temporary files and then a second layer with the clean up.

```suggestion
RUN dnf -y install libsecret libX11-devel libxkbcommon \
    "https://rpmfind.net/linux/centos/8-stream/BaseOS/x86_64/os/Packages/libsecret-devel-0.18.6-1.el8.x86_64.rpm" \
    "https://rpmfind.net/linux/centos/8-stream/AppStream/x86_64/os/Packages/libxkbfile-1.1.0-1.el8.x86_64.rpm" \
    "https://rpmfind.net/linux/centos/8-stream/PowerTools/x86_64/os/Packages/libxkbfile-devel-1.1.0-1.el8.x86_64.rpm" \
    "https://rpmfind.net/linux/centos/8-stream/BaseOS/x86_64/os/Packages/zsh-5.5.1-6.el8_1.2.x86_64.rpm" \
    util-linux-user && \
    dnf -y clean all --enablerepo='*'
```

---

Solved